packages/ui-library/src/components/icon-menu/index.tsx

```diff
@@ -3,13 +3,41 @@ import { ButtonProps, Button } from '../button';
 import { Menu } from '../menu';
 import { KeyboardArrowDownIcon } from '../../icons';
 import { MenuItem } from '../menuitem';
+import { Box } from '../box';
+import { Divider, Typography } from '@mui/material';
+import styled from 'styled-components';
+import { colors } from '../../theme';
+
+const StyledSecondaryLabel = styled(Typography)`
+  ${({ theme: { palette, typography } }) => `
+  color: ${colors[palette.mode].typography.typo3};
+  font-size: ${typography.bodyExtraSmall.fontSize};
+`}
+`;
+
+const StyledMenuItemContent = styled(Box)`
+  ${({ theme: { spacing } }) => `
+  display: flex;
+  align-items: center;
+  gap: ${spacing(2)};
+`}
+`;
+
+const StyledDivider = styled(Divider)`
+  ${({ theme: { spacing } }) => `
+  padding-top: ${spacing(2)};
```

Comment:
For this solution, the divider is contained in the same box as the MuiMenuItem, so it needs a made-up gap of 8 px represented by this padding-top. I believe <Divider /> is used in many places and it may not need this gap